### PR TITLE
Use consistent int-ll64 typedefs for aarch32 and aarch64

### PR DESCRIPTION
**Pull request by masahir0y**

### CODE REVIEW

**Comment by masahir0y** [comment]
This is related to a discussion in PR 1249.

Here is the repeat of the log of the last commit.

    types: use int-ll64 for both aarch32 and aarch64
    
    The result of commit 031dbb122472 ("AArch32: Add essential Arch
    helpers") is a nightmare.
    
    For example, 'uint64_t' is defined as 'unsigned long long' for aarch32
    and 'unsigned long' for aarch64.
    
    Likewise, 'uintptr_t' is 'unsigned int' for aarch32, 'unsigned long'
    for aarch64.
    
    This is a problem when you use printf() in common code.
    
    Use ifdef?
      #ifdef AARCH32
              printf("val = %llx", val);
      #else
              printf("val = %lx", val);
      #endif
    
    or cast to a type long enough to cover all cases?
      printf("val = %lx\n", (unsigned long long)val);
    
    or even uglier?
      printf("val = %" PRIx64 "\n", val);
    
    No.  Use the same typedefs for all arches.  As you see in Linux commit
    0c79a8e29b5f ("asm/types.h: Remove include/asm-generic/int-l64.h"),
    kernel developers completed the task to unify in-kernel typedefs into
    int-ll64, where fixed integer types can be defined as follows:
    
        typedef signed char           int8_t;
        typedef unsigned char         uint8_t;
    
        typedef signed short          int16_t;
        typedef unsigned short        uint16_t;
    
        typedef signed int            int32_t;
        typedef unsigned int          uint32_t;
    
        typedef signed long long      int64_t;
        typedef unsigned long long    uint64_t;
    
    This makes sense where the codebase is shared between 32bit and 64bit,
    with the data mode ILP32, LP64, respectively.

                       ILP32           LP64
        int            32              32
        long           32              64
        long long      64              64
    
    'long long' is always 64bit so used for defining 'uint64_t'.
    'long' has the same width as pointer, so for 'uintptr_t'.
    
    I left an ifdef conditional for (s)size_t because "%z" is used in some
    places.  (maybe we should change size_t to 'unsigned long' by forbidding
    the use of "%z")

I believe this is the right thing, and will make clean code-base in a long run.

However, this commit can not be applied until typedef distortion is fixed.

There are several mismatch between declaration in header and definition in C file.

I think I can do this work, but I am busy.
If somebody offers to do this tedious but valuable work,
I am happy to hand this over.

Note:
Nothing in this PR is signed-offed yet.

**Comment by antonio-nino-diaz-arm** [comment]
I think that the idea is good, but this is might cause problems for non-upstream platforms depending on where the changes have to be done. Let's wait for more comments.

**Comment by soby-mathew** [comment]
Yes, usage of `unsigned long`  was always problematic between AArch32 and AArch64. The original dilemma when the AArch32 mode was introduced was whether to modify the *standard* headers in `stdlib`. But that dilemma is passed now I think.

File: include/lib/stdlib/machine/_types.h [hunk 1]
```diff
@@ -110,8 +73,23 @@ typedef	__int8_t	__int_least8_t;
 typedef	__int16_t	__int_least16_t;
 typedef	__int32_t	__int_least32_t;
 typedef	__int64_t	__int_least64_t;
+typedef	long		__ptrdiff_t;		/* ptr1 - ptr2 */
+typedef	long		__register_t;
+typedef	long		__segsz_t;		/* segment size (in pages) */
+/* differentiate for "%z" in printf() */
+#ifdef	AARCH32
+typedef	unsigned int	__size_t;		/* sizeof() */
+typedef	int		__ssize_t;		/* byte count or error */
```
Comment (danh-arm) [inline include/lib/stdlib/machine/_types.h]: Perhaps we can fix tf_printf to always using long for %z rather than have the #ifdef here?

**Comment by dp-arm** [comment]
@masahir0y ping

**Comment by masahir0y** [comment]
Sorry, I did not notice this PR had been replied.
I will find some time to work on it next week.

**Comment by antonio-nino-diaz-arm** [comment]
IP Review +1

**Comment by antonio-nino-diaz-arm** [comment]
I would squash all "add static qualifier to locally used functions" and all "fix pointer type mismtach" commits. Also, you have a typo there: mismtach.

**Comment by masahir0y** [comment]
> I would squash all "add static qualifier to locally used functions" and all "fix pointer type mismtach" commits. Also, you have a typo there: mismtach.

OK.  I squashed them, and fixed the typo.

File: include/lib/stdlib/machine/_types.h (hunk 1 above)
Comment (danh-arm) [reply include/lib/stdlib/machine/_types.h]: Yes, good point.

**Comment by danh-arm** [comment]
jenkins: test this please

**Comment by masahir0y** [comment]
> Could you take a look at them?

I fixed them and pushed.
Thanks.

**Comment by antonio-nino-diaz-arm** [comment]
The Coverity tests have failed, but everything else has passed.

**Pull request merged**